### ISSUE TITLE
IC2 Classic

### ISSUE THREAD

**Issue by Speiger** [Unity-Resource-Pack/Unity#55]
Could you add IC2 Classic Support? It can be combined with IC2 Exp in Textures (it uses the same folder but differend subfolders)

**Comment by Yulife** [comment]
We'll keep it in mind :D

**Comment by Speiger** [comment]
1 Upside. Texture Sprites. You do not have to do 5000 Icons. Its all in a couple texture sprites merged together (at least in classic) xD

**Comment by BBoldt** [comment]
Would probably be best to use the same textures that we use for IC2-exp, just "ported" to whatever file/image structure IC2-Classic has. Then we would just need to fill in the blanks for the extra content that IC2 Classic has.

**Comment by Speiger** [comment]
@BBoldt Yeah but exp changed they general style and classic kept the old one

**Comment by Yulife** [comment]
Sprites have downsides too, for example if you do not have all textures done you either need to fill in the not textured with blanks or ship the author's textures which may conflict with some licences :P But yea, sprites are nice

**Comment by Speiger** [comment]
yeah that is true. But on the other hand. As long you finish your work and make it a "none constant" state it should be fine (at least from my side)

**Comment by Yulife** [comment]
Just curious, IC2 Classic and Tech Reborn are completly different right? Because we started on Tech Reborn! :smile:

**Comment by BBoldt** [comment]
They are Yulife. Tech Reborn is a rewritten version of an old version of Gregtech. IC2 Classic is an updated version of pre-experimental IC2.

**Comment by Yulife** [comment]
Ahhh nice :)

**Comment by CyanideX** [comment]
@Speiger, you still looking to get those textures for IC2C? I'm just about done with IC2 so it wouldn't be too difficult to throw something together.

**Comment by Yulife** [comment]
Shouldn't it use them same texture? :p

**Comment by Speiger** [comment]
@CyanideX well it would be great if IC2C gets texture pack support...
@Yulife well thats your choice but IC2Exp did start a new Art Style while classic tries to keep the old one... So if you find something that fits for both then i am fine with that.

**Comment by Yulife** [comment]
Hey @CyanideX once you get IC2 done I can take a look at IC2 Classic's sprites, just tell me once it's done

**Comment by Mattabase** [comment]
@CyanideX @Yulife Would super dig those IC2C textures <3

**Comment by CyanideX** [comment]
@Mattabase, I will see what I can do.

**Comment by Mattabase** [comment]
<3 Love you buddy! @CyanideX =P

**Comment by CyanideX** [comment]
@Mattabase, don't thank me yet. I may not get to it for a while. 😖

**Comment by CyanideX** [comment]
Sorry, can't do it @Mattabase. I would need @Speiger to make the bottom/top face rotate with the block in order for the textures to work properly. This retexture would be a basic copy/paste from my IC2 Unity because I simply don't have the time to make adjustments for this system. 

![Sample](http://i.imgur.com/eRFeYie.jpg)

**Comment by Speiger** [comment]
@CyanideX the problem is i would require my full renderring for that just making the stuff work.
I am using the standart renderer.... hmmmm i didnt expect that you do side awarenes...

**Comment by Speiger** [comment]
@CyanideX ok i will try to come up with something. But might take a couple days...

**Comment by Speiger** [comment]
@CyanideX keep doing the work even rotation is not applying yet. I found some interesting code...
That should fix that issue very fast... (Patch will come out soon)
@Mattabase notify xD

**Comment by CyanideX** [comment]
@Speiger, I tend to get fancy with my textures. 😉 Thanks for looking into this.

**Comment by CyanideX** [comment]
It's coming along:
![Progress!](http://i.imgur.com/owIY2Zt.png)

**Comment by Speiger** [comment]
Cool :3 i wonder how you do the jetpacks etc xD because they have no textures on Exps Side

**Comment by Speiger** [comment]
@CyanideX do you want to add animations? I can do a oneliner code which allows you to make animations to IC2Classic without using the sprite system while stay still compatible^^"

**Comment by CyanideX** [comment]
@Speiger, the jetpacks had textures but here are some of the items from my retexture:

![Items](http://i.imgur.com/DOQ2E4B.jpg)

I didn't do any animations aside from a few items so no need to worry about that.

**Comment by CyanideX** [comment]
Ok, @Speiger, I'm going to need some animations for a few items.

**Comment by CyanideX** [comment]
We're getting there:

![image](http://i.imgur.com/zGZcqAr.png)

Nothing a night of beer can't accomplish.

**Comment by Speiger** [comment]
Well I can't tell you where to put it now but,
Leave the texture in the sprite that is going to be animated like it is. Make a separate file that is like vanilla Minecraft texture and do the animation there.
Simply: The texture loader from class inject 2 paths into the texture. 1: the path of the sprite + coords. 2: a replacement path + name which loads if he finds the texture there.
if the loader finds it there it disables classic loading kind and loads the texture normal like any vanilla mc texture.
(Including mc meta files) 
I can tell you the name as soon as I am in my dev environment.

**Comment by Speiger** [comment]
@CyanideX well ok i did add the code for the Texture Replacing into my dev invioment (will be with the next release)
So the Textures you want to replace with special Properties you need to place into this path:
"ic2:textures/replacements/(SpriteID)X(XPosition)Y(YPosition).png"
for example you want to replace a the Nightvision Goggles the file of the texture has to be called like this:
"i2X2Y2.png" (its a weird name i know but thats the best system i came up with)

Now working on the Rotation part and last the Second Texture for the Nuke.
Also here is a list of Sprite to ID:
```
"b0" = "ic2:textures/sprites/block_0.png"
"bc" = "ic2:textures/sprites/crops_0.png"
"bgen" = "ic2:textures/sprites/block_generator.png"
"bmach" = "ic2:textures/sprites/block_machine.png"
"bmach2" = "ic2:textures/sprites/block_machine2.png"
"bpersonal" = "ic2:textures/sprites/block_personal.png"
"bcable" = "ic2:textures/sprites/block_cable.png"
"belec" = "ic2:textures/sprites/block_electric.png"
"i0" = "ic2:textures/sprites/item_0.png"
"i1" = "ic2:textures/sprites/item_1.png"
"i2" = "ic2:textures/sprites/item_2.png"
"i3" = "ic2:textures/sprites/item_3.png"
"batBox" = "ic2:textures/sprites/Batbox.png"
"mfe" = "ic2:textures/sprites/MFE.png"
"mfsu" = "ic2:textures/sprites/MFSU.png"
"bBox" = "ic2:textures/sprites/BatteryBox.png"
"triggers" = "ic2:textures/sprites/bc_triggers.png"
"conv" = "ic2:textures/sprites/block_Converter.png"
"CPad" = "ic2:textures/sprites/block_pads.png"
```

**Comment by Speiger** [comment]
@CyanideX version is out. You find it on github... Version 1.2.1.2.3

**Comment by CyanideX** [comment]
Hey @Speiger, I downloaded the new version, created the replacements directory, adding the animation .png as well as the meta file, named it i3x3y1.png to target the overclocked heat vent, but it doesn't appear to do anything.

![dir](http://i.imgur.com/FXEYVx0.png)

**Comment by Speiger** [comment]
@CyanideX i made a mistake... Its not ic2/textures/replacements. its ic2/replacements.

**Comment by Speiger** [comment]
it should not colide with ic2Exp at all.. So i think i can leave it like that^^"

**Comment by CyanideX** [comment]
@Speiger, moved them to ic2/replacements and it still doesn't seem to be working. 😖

**Comment by Speiger** [comment]
@CyanideX i make a hotfix for my derpiness...

**Comment by Speiger** [comment]
put them back to the ic2/textures/replacements

**Comment by Speiger** [comment]
found another bug with plugins so i can make a fast release.... Third today xD

**Comment by Speiger** [comment]
its now out @CyanideX

**Comment by Speiger** [comment]
https://github.com/TinyModularThings/IC2Classic/releases/tag/1.2.1.2.4.1

**Comment by Speiger** [comment]
As soon you say its ok i put it on curse....

**Comment by Speiger** [comment]
(dont want a forth release today xD

**Comment by CyanideX** [comment]
@Speiger, that's a lot of posts you've done! 😉 Maybe editing one post would be better.

Also, I tried the new version and the animations still don't seem to be working. I put them back into ic2/textures/replacements.

**Comment by Speiger** [comment]
i2X2Y2.png everything has to be typed big.

**Comment by CyanideX** [comment]
@Speiger:

![dir](http://i.imgur.com/x5d1iiJ.png)

![ingame](http://i.imgur.com/p9d4ABy.png)

**Comment by Speiger** [comment]
@CyanideX could you join the IRC channel then i do not have to do releases....
(now its try and errors) (IC2-Classic is the channel)

**Comment by Yulife** [comment]
@Speiger you use JSONs? Also why the weird PNG names?

**Comment by Speiger** [comment]
@Yulife Nope.
Why the weird name? Because thats how the Sprites are loaded. Its cutting the Textures into pieces ingame so that do not have to place 1k Textures into my folder... It makes replacing basic stuff very easy while keeping special stuff like animations in a special area and allow you to remove/edit them very easy.

**Comment by CyanideX** [comment]
@Speiger, I **_love_** working with sprite sheets because it allows me to make changes to all of the textures with a single adjustment layer instead doing them to 256 individual textures; however, the one downside is trying to figure out what the items are without seeing their filename. I found myself going in game trying to match the texture with some of the items/blocks to know what I should be drawing. :P

**Comment by Speiger** [comment]
hehehe xD

**Comment by Speiger** [comment]
how is progress going @CyanideX ?

**Comment by CyanideX** [comment]
@Speiger, the retexture is on hiatus until I get a few other projects out of the way. Work is keeping me busy.

**Comment by Speiger** [comment]
ok. It could be possible that you can fast update 1.7.10 to 1.10.2 xD When the speed stays like this xD

**Comment by Mattabase** [comment]
Poke @CyanideX

**Comment by CyanideX** [comment]
Damnit, @Mattabase... I was hoping your were intoxicated enough to forget. :P

**Comment by Speiger** [comment]
Hahaha I think mattabase is someone who doesn't forget...
Also @Mattabase & @CyanideX i just recently added the 1.7.10 Texture rotation feature to 1.10.2 xD that was easier to add then I expected xD

**Comment by Mattabase** [comment]
Nope, I remember @CyanideX =P

**Comment by Mattabase** [comment]
Poke @CyanideX Just your friendly neighborhood Matta sending you a poke. =P

**Comment by CyanideX** [comment]
Alright, I will get the rest of these finished soon! I didn't forget.

**Comment by Mattabase** [comment]
Wanders in... @CyanideX

**Comment by TheCodedOne** [comment]
_Wanders out..._

**Comment by Speiger** [comment]
@Mattabase i doubt @CyanideX will finish it how you would like to have it because the mod grew already about an extra 100% between 1.7.10 and 1.10.2 the best you will get is maybe 1.7.10 and thats it.
On top of that IC2C dies the moment 1.10.2 is no longer played.

**Comment by InsomniaKitten** [comment]
Why is IC2C going to die on newer versions past 1.10.2? Is it lack of free time to port it?

**Comment by Speiger** [comment]
@InsomniaKitten its because i want to leave minecraft modding.
1.11.2 Would be another 4 months of development when i am done 1.12 is the main version.
And 1.11.2 is basically a 40% ModRewrite because anything with Items related which is basically in almost everything in the mod.
And 1.12 is something that removes TileEntities which IC2C is based of. So i can scrap everything and start from new. And i kinda dont want to have my time wasted on 1.10.2 because i put a lot of effort into it. 
On top of that i cant play/mod in 1.12 because my GPU (PC) is so much a piece of shit that it doesnt support the new OpenGL Version that comes with 1.12. I would not be able to run minecraft at any form. And thats a requirement to design a good mod.
And there is some different reasons with IC2Exp, Minecraft and Forge in its entirety that want me to leave. Basically Minecraft&Forge make the envioment for a developer hostile. If the developer cares about its work more then about the money income. And the IC2Exp Chains i have making it not better.

**Comment by TehNut** [comment]
>1.12 is something that removes TileEntities 

There is literally no evidence of this in the snapshot. *Maybe* 1.13, but probably even later.

**Comment by Speiger** [comment]
@TehNut that doesnt change the fact of a lot of different changes.

**Comment by InsomniaKitten** [comment]
If someone were to port it, would you be willing to continue maintaining it?

**Comment by Speiger** [comment]
@InsomniaKitten nobody can port it since ic2c is closed source and nobody except me gets the license for IC2Classic. 
And i can ensure you this: As soon it gets out of my hand and someone tries to port it the quallity drops drastilcly.
IC2Classic is the most unatractive thing you can take on in terms if you want to get money from it.
Its a project that takes a lot of time, which people will flame because its different, will lose different modsupport because it does not use the HypeTrain systems (which are very laggy in the first place),
and you have to deal on top of that with what shit IC2Exp brings out as API. (not joking).
And then there comes one thing. If player accepts the port (which i doubt) then you have to give away a very big chunk of the money to him if you want to put it on curse (that you would get from curse)
So you are basically making it to produce money for someone else.

Thats why its unatractive for other modders.
And here is the reason why it would lose quallity if someone else takes it over.
Do you know a modder that invests 4-8 hours on a dayly bases for 1 mod for 1 year?
(weekends included so no break there) i had rarely a day or 2 free. The longest break i ever had was 2 weeks (that was before i started and when my pc broke). I would count that i would have maybe 1-2 months in total free overall in that 1 year.
There is a lot of time i invested into this thing to make it as good as my tester like it.
1.10.2 version will be like a extreme polishment of 1.7.10 + a lot of additions.
And i doubt that people would do that with this low of Reward for them...

**Comment by CyanideX** [comment]
Soooo... should I finish it?

**Comment by Speiger** [comment]
@CyanideX you can finish it. I think people would love to have it as long  they play in 1.10.2 & 1.7.10

**Comment by Mattabase** [comment]
Poke @CyanideX So, how about those textures?

**Comment by CyanideX** [comment]
I can finish them I suppose.

It's too bad it won't be for 1.11+... feels like wasted time. But I did promise you I would.

**Comment by CyanideX** [comment]
Just want to confirm, @Speiger... you're not updating IC2C for 1.12? I think I misunderstood your Reddit post.

**Comment by Speiger** [comment]
So far nope.
This 13 months have exhausted me from mc modding and the issue is in that type of area you have no time to break as long as I would need...

**Comment by Mattabase** [comment]
Should close this issue, mod requested is dead.

**Comment by CyanideX** [comment]
I let you down Matt, I'm sorry.

**Comment by Mattabase** [comment]
No let down sir. <3

**Comment by CyanideX** [comment]
Will reopen if this ever gets updated to a current Minecraft version.

**Comment by CyanideX** [comment]
The day has come.
